lib/url.c

```diff
@@ -5689,6 +5704,8 @@ static void reuse_conn(struct connectdata *old_conn,
   old_conn->recv_pipe = NULL;
 
   Curl_safefree(old_conn->master_buffer);
+
+  Curl_safefree(old_conn->unix_domain_socket);
```

Comment:
Needs ifdef

(Ok, this is indeed correct. `old_conn` is the (pointer to the) structure that will be replaced by `conn`, so to avoid resource leaks it must be freed here.)